### DISCUSSION TITLE
session dose not exist in test but in action it dose

### DISCUSSION THREAD

**Discussion by FzSalehi** [laravel/framework#41292] [Q&A]
Hi  , im using  this [captcha]([mewebstudio](https://github.com/mewebstudio) /
[captcha](https://github.com/mewebstudio/captcha)).
in action , if i `dd(sesstion->all())` in view component , i have access to session created by this packege.
```
array:8 [[▼]()
  "_token" => "OFUim4KFT1Ezhea4NrAK5I8FfXHg98ld8PLM5uVk"
  "national_id" => "0010432213"
  "personnelCode" => "10432213"
  "mobile" => "09124181585"
  "type" => "staff"
  "_flash" => array:2 [[▶]()]
  "_previous" => array:1 [[▶]()]
  "captcha" => array:3 [[▼]()
    "sensitive" => false
    "key" => "eyJpdiI6Im5BV3UyM2duMk0zb0pML0cvaDFFVWc9PSIsInZhbHVlIjoiWm9uM0lpZDlxQk9vY1FNNGUrRmZzN0tiMnZGTlY3U2ZZcGhoYlFMWkMwREYrdE9zbFdRTUI3b1Jjdko1SWpaRFhZQzNJdlprMWV2UHly[ ▶]()"
    "encrypt" => true
  ]
]
```

but if i die and dump session  in test after this page has loaded  .. i wont be able to access to session created by this package.
this is mytest :
```
/**
   * @test
   * @return void
   */
  public function a_user_can_request_for_registration_with_correct_data(): void
  {
    // Enter to Register page to create sessions

    $response = $this->get('/register/temp-user/staff');


    dd(session()->all());
  }
```
and this is session that shows :
```
array:3 [
  "_token" => "sC62lqkfkQPDR7LP6ZEn4EmnIt6wiyQBIY7Xm4jZ"
  "_previous" => array:1 [
    "url" => "https://register.khu.ac.ir/register/temp-user/staff"
  ]
  "_flash" => array:2 [
    "old" => []
    "new" => []
  ]
]
// No Captcha Session Here !
```
Im looking a way to pass this captcha in feature tests ... if any solution you  can suggest .. feel free and share with me .. im about 3 day trying to pass this test ...

ps. sorry for my bad english.

**Comment by gguney** [comment]
Maybe you can mock the captcha rather than using it directly
